### PR TITLE
CLI tool subcommand to show failed test cases from a claim file.

### PR DESCRIPTION
**Pull request by greyerof**
Added a new subcommand to the tnf cli tool that parses a claim file and
shows the failed test cases in a more readable way. By default, the
output is plain text, but a 'json' output mode can be selected in order
to get a json representation of those failed test cases per test suite.

Also, the subcommand has a flag --testsuites that can be used to reduce
the output to the tests suites included in the comma separated list.

**Examples:**
1. Show all the failed test cases from all test suites:
./tnf claim show failures --claim path/to/claim.json

2. Show all the failed test cases from all tests suites in json format.
./tnf claim show failures --claim path/to/claim.json --output json

3. Show all the failed test cases from the lifecycle and access-control
   suites.
./tnf claim show failures --claim path/to/claim.json --testsuites
"lifecycle,access-control"

**Plain text output sample:**
```
./tnf claim show failures --claim path/to/claim.json
Test Suite: access-control
  Test Case: access-control-sys-admin-capability-check
    Description: Ensures that containers do not use SYS_ADMIN capability
    Failure reasons:
       1 - Type: Container, Reason: Non compliant capability detected in container
           Namespace: tnf, Pod Name: test-887998557-8gwwm, Container Name: test, SCC Capability: SYS_ADMIN
       2 - Type: Container, Reason: Non compliant capability detected in container
           Namespace: tnf, Pod Name: test-887998557-pr2w5, Container Name: test, SCC Capability: SYS_ADMIN
  Test Case: access-control-security-context
    Description: Checks the security context matches one of the 4 categories
    Failure reasons:
       1 - Type: ContainerCategory, Reason: container category is NOT category 1 or category NoUID0
           Namespace: tnf, Pod Name: jack-6f88b5bfb4-q5cw6, Container Name: jack, Category: CategoryID4(anything not matching lower category)
       2 - ...
       ...
Test Suite: lifecycle
  Test Case: ...
    ...
 ```
 **JSON output sample:**
 ```
 ./tnf claim show failures --claim path/to/claim.json --output json
 {
  "testSuites": [
    {
      "name": "access-control",
      "failures": [
        {
          "name": "access-control-sys-admin-capability-check",
          "description": "Ensures that containers do not use SYS_ADMIN capability",
          "nonCompliantObjects": [
            {
              "type": "Container",
              "reason": "Non compliant capability detected in container",
              "spec": {
                "Namespace": "tnf",
                "Pod Name": "test-887998557-8gwwm",  
                "Container Name": "test",
                "SCC Capability": "SYS_ADMIN"
              }
            },
            {
              "type": "Container",
              "reason": "Non compliant capability detected in container",
              "spec": {
                "Namespace": "tnf",
                "Pod Name": "test-887998557-pr2w5",  
                "Container Name": "test",
                "SCC Capability": "SYS_ADMIN"
              }
            }
          ]
        },
      ]
    },
    {
      "name" : "lifecycle",
      "failures": [
          ...
          ...
      ]
    },
  ]
}
 ```
 The first commit (adff9b87bb6e12d4ed9d73e752dbba8b5b3c548b) in this PR is just a reorganization of the
existing files, as it was a bit messy. Every command/subcommand has
its own folder now. This is the current subcommands tree:
```
tnf
|- claim
|  |- add
|  |- compare
|  |- show
|     |- failures
|- generate  
   |- catalog
   |- javascript
   |- markdown
```

### CODE REVIEW

File: cmd/tnf/claim/compare/compare.go [hunk 1]
```diff
@@ -7,30 +7,39 @@ import (
...
+)
+
+func NewCommand() *cobra.Command {
+	claimCompareFiles.Flags().StringVarP(
+		&Claim1, "claim1", "1", "",
+		"existing claim1 file. (Required) first file to compare",
+	)
+	claimCompareFiles.Flags().StringVarP(
+		&Claim2, "claim2", "2", "",
+		"existing claim2 file. (Required) second file to compare with",
```
Comment (sebrandon1) [inline cmd/tnf/claim/compare/compare.go:31]:
```suggestion
		"existing claim2 file. (Required) second file to compare",
```

Match the other string?

File: cmd/tnf/claim/compare/compare.go [hunk 2]
```diff
@@ -7,30 +7,39 @@ import (
...
+	)
+	err := claimCompareFiles.MarkFlagRequired("claim1")
+	if err != nil {
+		return nil
```
Comment (sebrandon1) [inline cmd/tnf/claim/compare/compare.go:35]: Should we surface the errors?

**Review by sebrandon1 — COMMENTED** [review]
Looks great, just some initial comments!

File: cmd/tnf/claim/show/failures/failures.go [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+
+// Main function for the `show failures` subcommand.
+func showFailures(_ *cobra.Command, _ []string) error {
+	outputFormat, err := parseOutputFormatFlag()
+	if err != nil {
+		return err
+	}
+
+	// Parse the claim file into the claim scheme.
+	claimScheme, err := claim.Parse(claimFilePathFlag)
```
Comment (edcdavid) [inline cmd/tnf/claim/show/failures/failures.go:276]: Optional(could be another PR): Add a claim format check before trying to decode the claim file

File: cmd/tnf/claim/compare/compare.go (hunk 2 above)
Comment (greyerof) [reply cmd/tnf/claim/compare/compare.go:35]: Added an error trace.

File: cmd/tnf/claim/compare/compare.go (hunk 1 above)
Comment (greyerof) [reply cmd/tnf/claim/compare/compare.go]: Fixed in https://github.com/test-network-function/cnf-certification-test/pull/1249/commits/0d669a36eddc5a5e186fed2ccbb63390082b84ff. Please note that most of the code of the compare subcommand was left as it was before. I just made the minimum changes to adapt the command/subcommand tree loading to the new folders structure.

**Review by sebrandon1 — APPROVED** [review]
